File: server/test/java/com.bomberman/utils/CollisionsTest.java
package com.bomberman.utils;

import com.bomberman.TestDataGenerator;
import com.bomberman.entities.mapobject.movable.Player;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class CollisionsTest {

  @Test
  void willCollide_playerAboveObstacle_moveDown() {
    final TestDataGenerator generator =
        TestDataGenerator.newGenerator()
            .createPlayerAtColRow("player1", 7, 5)
            .createWallAtColRow(7, 6);

    final Player player = generator.getPlayer("player1");

    Assertions.assertTrue(
        Collisions.willCollide(
            player,
            TestDataGenerator.positionAtColRow(7, 6),
            generator.getGameState().getObstacles()));
  }

  @Test
  void willCollide_playerAboveObstacle_moveUp() {
    final TestDataGenerator generator =
        TestDataGenerator.newGenerator()
            .createPlayerAtColRow("player1", 7, 5)
            .createWallAtColRow(7, 6);

    final Player player = generator.getPlayer("player1");

    Assertions.assertFalse(
        Collisions.willCollide(
            player,
            TestDataGenerator.positionAtColRow(7, 4),
            generator.getGameState().getObstacles()));
  }

  @Test
  void willCollide_playerAboveObstacle_moveLeft() {
    final TestDataGenerator generator =
        TestDataGenerator.newGenerator()
            .createPlayerAtColRow("player1", 7, 5)
            .createWallAtColRow(7, 6);

    final Player player = generator.getPlayer("player1");

    Assertions.assertFalse(
        Collisions.willCollide(
            player,
            TestDataGenerator.positionAtColRow(6, 5),
            generator.getGameState().getObstacles()));
  }

  @Test
  void willCollide_playerAboveObstacle_moveRight() {
    final TestDataGenerator generator =
        TestDataGenerator.newGenerator()
            .createPlayerAtColRow("player1", 7, 5)
            .createWallAtColRow(7, 6);

    final Player player = generator.getPlayer("player1");

    Assertions.assertFalse(
        Collisions.willCollide(
            player,
            TestDataGenerator.positionAtColRow(8, 5),
            generator.getGameState().getObstacles()));
  }

  @Test
  void willCollide_playerUnderObstacle_moveDown() {
    final TestDataGenerator generator =
        TestDataGenerator.newGenerator()
            .createPlayerAtColRow("player1", 7, 5)
            .createWallAtColRow(7, 4);

    final Player player = generator.getPlayer("player1");

    Assertions.assertFalse(
        Collisions.willCollide(
            player,
            TestDataGenerator.positionAtColRow(7, 6),
            generator.getGameState().getObstacles()));
  }

  @Test
  void willCollide_playerUnderObstacle_moveUp() {
    final TestDataGenerator generator =
        TestDataGenerator.newGenerator()
            .createPlayerAtColRow("player1", 7, 5)
            .createWallAtColRow(7, 4);

    final Player player = generator.getPlayer("player1");

    Assertions.assertTrue(
        Collisions.willCollide(
            player,
            TestDataGenerator.positionAtColRow(7, 4),
            generator.getGameState().getObstacles()));
  }

  @Test
  void willCollide_playerUnderObstacle_moveLeft() {
    final TestDataGenerator generator =
        TestDataGenerator.newGenerator()
            .createPlayerAtColRow("player1", 7, 5)
            .createWallAtColRow(7, 4);

    final Player player = generator.getPlayer("player1");

    Assertions.assertFalse(
        Collisions.willCollide(
            player,
            TestDataGenerator.positionAtColRow(6, 5),
            generator.getGameState().getObstacles()));
  }

  @Test
  void willCollide_playerUnderObstacle_moveRight() {
    final TestDataGenerator generator =
        TestDataGenerator.newGenerator()
            .createPlayerAtColRow("player1", 7, 5)
            .createWallAtColRow(7, 4);

    final Player player = generator.getPlayer("player1");

    Assertions.assertFalse(
        Collisions.willCollide(
            player,
            TestDataGenerator.positionAtColRow(8, 5),
            generator.getGameState().getObstacles()));
  }
}

File: server/src/test/kotlin/com/bomberman/game/CollisionsTest.kt
package com.bomberman.game

import com.bomberman.TestDataGenerator
import com.bomberman.positionAtColRow
import org.junit.jupiter.api.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class CollisionsTest {

  @Test
  fun willCollide_playerAboveObstacle_moveDown() {
    val generator = TestDataGenerator()
        .createPlayerAtColRow("player1", 7, 5)
        .createWallAtColRow(7, 6)

    val player = generator.getPlayer("player1")

    assertTrue { player.willCollide(positionAtColRow(7, 6), generator.gameState.obstacles) }
  }

  @Test
  fun willCollide_playerAboveObstacle_moveUp() {
    val generator = TestDataGenerator()
        .createPlayerAtColRow("player1", 7, 5)
        .createWallAtColRow(7, 6)

    val player = generator.getPlayer("player1")

    assertFalse { player.willCollide(positionAtColRow(7, 4), generator.gameState.obstacles) }
  }

  @Test
  fun willCollide_playerAboveObstacle_moveLeft() {
    val generator = TestDataGenerator()
        .createPlayerAtColRow("player1", 7, 5)
        .createWallAtColRow(7, 6)

    val player = generator.getPlayer("player1")

    assertFalse { player.willCollide(positionAtColRow(6, 5), generator.gameState.obstacles) }
  }

  @Test
  fun willCollide_playerAboveObstacle_moveRight() {
    val generator = TestDataGenerator()
        .createPlayerAtColRow("player1", 7, 5)
        .createWallAtColRow(7, 6)

    val player = generator.getPlayer("player1")

    assertFalse { player.willCollide(positionAtColRow(8, 5), generator.gameState.obstacles) }
  }

  @Test
  fun willCollide_playerBelowObstacle_moveDown() {
    val generator = TestDataGenerator()
        .createPlayerAtColRow("player1", 7, 5)
        .createWallAtColRow(7, 4)

    val player = generator.getPlayer("player1")

    assertFalse { player.willCollide(positionAtColRow(7, 6), generator.gameState.obstacles) }
  }

  @Test
  fun willCollide_playerBelowObstacle_moveUp() {
    val generator = TestDataGenerator()
        .createPlayerAtColRow("player1", 7, 5)
        .createWallAtColRow(7, 4)

    val player = generator.getPlayer("player1")

    assertTrue { player.willCollide(positionAtColRow(7, 4), generator.gameState.obstacles) }
  }

  @Test
  fun willCollide_playerBelowObstacle_moveLeft() {
    val generator = TestDataGenerator()
        .createPlayerAtColRow("player1", 7, 5)
        .createWallAtColRow(7, 4)

    val player = generator.getPlayer("player1")

    assertFalse { player.willCollide(positionAtColRow(6, 5), generator.gameState.obstacles) }
  }

  @Test
  fun willCollide_playerBelowObstacle_moveRight() {
    val generator = TestDataGenerator()
        .createPlayerAtColRow("player1", 7, 5)
        .createWallAtColRow(7, 4)

    val player = generator.getPlayer("player1")

    assertFalse { player.willCollide(positionAtColRow(8, 5), generator.gameState.obstacles) }
  }

}